l3_dump.py

```diff
@@ -46,28 +55,11 @@
     READELF_DATA_SECTION = '__cstring'
 
 # #############################################################################
-def set_decode_loc_id(program_bin:str, loc_decoder_bin:str):
-    """
-    Check for required LOC-decoder binary, based on environment of run.
-    If L3_LOC_ENABLED=1 is set in the env:
-     - Search for a LOC-decoder binary specified by the --loc-binary argument.
-     - Otherwiwse, search for a default LOC-binary named "<program_binary>_loc".
-    """
-    decode_loc_id = DECODE_LOC_ID
-    loc_decoder = LOC_DECODER
-    if LOC_ENABLED in os.environ:
-        env_var_val = os.getenv(LOC_ENABLED)
-        if env_var_val == "1":
-            loc_decoder = program_bin + "_loc" if loc_decoder_bin is None else loc_decoder_bin
-            if os.path.exists(loc_decoder) is False:
-                print(f"Env-var {LOC_ENABLED}=1 is set, but required "
-                      + f"LOC-decoder binary {loc_decoder} is not found.")
-                sys.exit(1)
-
-            # LOC is in-use and LOC-decoder binary was found.
-            decode_loc_id = 1
-
-    return (decode_loc_id, loc_decoder)
+# Status flags defined in src/l3.c for L3_LOG()->flags field
+L3_LOG_FLAGS_LINUX                = int('0x0001', 16)
+L3_LOG_FLAGS_MACOSX               = int('0x0002', 16)
+L3_LOG_FLAGS_LOC_ENCODING         = int('0x0010', 16)
+L3_LOG_FLAGS_LOC_ELF_ENCODING     = int('0x0020', 16)
```

Comment:
Qs. Is there a more Pythonic way of definining these status flags?

This is the best I could come up with ... suggestions welcome.

Also, fyi -- this creates a dependency between these bits and their definitions in `l3.c`.

In future revs, I will look into having one set of definitions in some `.h` file and source them somehow in this `py` script.

Reply:
Looks Pythonic enough to me (although I'm not really an expert here.

However, I wonder whether a bit-mask of flags is the most appropriate here -- it cannot be Linux _and_ MacOS. So maybe there should be an OS code (0 = Linux, 1 = MacOS, 2 = Windows, etc), and separately an encoding type (0 = no LOC, 1 = regular LOC, 2 = ELF LOC). Rather than storing in a single bitmask.

(Just a thought, I don't feel super strongly about it.)

Reply:
Should we also store the number of log entry slots while we're here?

Reply:
@gregthelaw -- I thought about this ... And You are right on both accounts.

I was just trying to save some precious log-header bytes so came up with bit-masks.

But, I agree, that's also potentially misleading as the platform bits are mutually-exclusive, just as are the LOC-encoding scheme bits.

Here is what I will do:

```
106 typedef struct l3_log
107 {
108     uint64_t        idx;
109     uint64_t        fbase_addr;
110     uint32_t        pad0;
111     uint16_t        ~pad2~ log_size;     // In # of log-entries; i.e. L3_MAX_SLOTS
112     ~l3_log_flags_t  flags;~
         BYTE              platform;
         BYTE.             loc_type;
113     uint64_t        pad1;
114     L3_ENTRY        slots[L3_MAX_SLOTS];
115 } L3_LOG;
```

----

I also thought of more compact representations, squeezing 4-bits for each `platform` and `loc_type` into 1-byte and then 1-byte to store `L3_MAX_SLOTS` as # of bits. This will save some bytes, but will restrict # of log-entries to be power-of-2. I don't immediately see a strong reason for this restriction.

Let's go with slightly liberal space-layout and can fine-tune as more development converges.